int16 mode
string[] color